plugins/emoji/plugin.js

```diff
@@ -630,6 +630,11 @@
 							// Comparing lowercased strings, because emoji should be case insensitive (#2167).
 							return item.id.toLowerCase().indexOf( emojiName ) !== -1;
 						} );
+					data = arrTools.map( data, function( item ) {
+						return CKEDITOR.tools.extend( item, {
```

Comment:
Extending objects in this place secretly modifies data model kept in `emojiList`.
What results that after first search some elements in emojiList will gain new key `name`. This create inconsistency in data structure.
What's more addition of `name` key will happen multiple times, when you search for the same name again and again. This might impact on performance a little bit, because extend method is run ever single time.

Because of reasons I suggest to modify directly data model after reading it from JSON or consult it with @Comandeer, @mlewand, maybe will have some better idea for this case?

Reply:
It should happen just once, that's for sure (I mean once for every result). For sure you don't want to change their ID during the runtime.

Also, if possible, I'd defer the conversion only when needed. So that you only make the conversion only for results that are now required. There's no point in converting 4k strings if you display just 10 of them...

So I'm fine with adding a new prop once, or keeping a separate dictionary in memory with converted entries.